### PR TITLE
Alternative Automatic MobiusController lifecycle management in `UIViewController`s

### PR DESCRIPTION
**Pull request by jeppes**
This is an alternative to https://github.com/spotify/Mobius.swift/pull/131

In this version, the MobiusController will be started when the view is visible and stopped when the view disappears.

### CODE REVIEW

File: MobiusExtras/Test/MobiusViewControllerTests.swift [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+import Foundation
+import MobiusCore
+import MobiusExtras
+import Nimble
+import Quick
+
+final class MobiusViewControllerTests: QuickSpec {
+    override func spec() {
+        context("View Controller using Mobius") {
```
Comment (jeppes) [inline MobiusExtras/Test/MobiusViewControllerTests.swift]: These tests don't actually test that calls to `viewDidAppear` and `viewDidDisappear` are forwarded. I could not think of a way to do this.

**Comment by rastersize** [comment]
@jeppes I’d much prefer the original approach with an ObjC associated object over this approach with a child VC and added subview. Since the latter approach has a higher performance and memory cost. As well as a potential for added confusion when debugging some UI (“where did this view come from” and “erf why is there a view covering all of my UI”). As well as the, while small, added risk that it interferes with some parts of the UI.

**Comment by jeppes** [comment]
@rastersize I agree. I don't know how much value there is in automating the call to stop the loop.
The best solution might be to create a controller similar which just supports starting and stopping and use the approach in #131 to attach it to a controller. Then we could document that it should be started/stopped in viewDidAppear/viewDidDisappear respectively.
It wouldn't be too big of a deal if it weren't stopped, since at least the ViewController would not be leaked.

File: MobiusExtras/Test/MobiusViewControllerTests.swift (hunk 1 above)
Comment (JensAyton) [reply MobiusExtras/Test/MobiusViewControllerTests.swift]: That seems proper, testing that UIKit works as documented isn’t a unit test. :-)

**Comment by JensAyton** [comment]
I feel quite strongly that best practice should be to `start` in `viewWillAppear` and `stop` in `viewDidDisappear`. However, as far as I can see this can’t be automated without messing with the view hierarchy or class swizzling (I can’t see a way to do it with notifications or KVO), and that’s probably too high a cost for a blessed standard pattern.

Right now I’m leaning towards the approach in #131 with documented best practices (and of course we should have a public tutorial which covers it, as well as GP). Note that best practices for a MobiusController that actually uses starty-stoppy behaviour includes having an `initiate` that properly gathers state from any stateful collaborators.

On the other hand, if we do it that way, is it actually any similar than having guidance about how to use `MobiusController` from a VC? One possible benefit I see is that we could add an opt-in protocol that hides the `Connectable` by making `receiveModel` a protocol method and `dispatchEvent` a protocol extension method.

**Comment by jeppes** [comment]
I think we're all in agreement that we don't want this approach, so I will close this PR.

**Pull request closed without merging**